src/java.desktop/share/classes/javax/swing/ImageIcon.java

```diff
@@ -321,18 +321,24 @@ public ImageIcon() {
      */
     protected void loadImage(Image image) {
         MediaTracker mTracker = getTracker();
+        boolean interrupted = false;
         synchronized(mTracker) {
             int id = getNextID();
 
             mTracker.addImage(image, id);
             try {
                 mTracker.waitForID(id, 0);
             } catch (InterruptedException e) {
-                System.out.println("INTERRUPTED while loading Image");
+                interrupted = true;
             }
+
             loadStatus = mTracker.statusID(id, false);
             mTracker.removeImage(image, id);
 
+            if (interrupted && ((loadStatus & MediaTracker.LOADING )!= 0)) {
```

Comment:
```suggestion
            if (interrupted && ((loadStatus & MediaTracker.LOADING ) != 0)) {
```
There should be one space on either side of a binary operator.

Reply:
There is a change of Behavior if its interrupted, for Not-Interrupted case the result would be ERRORED for url link and COMPLETE for local Image, but for Interrupted case the result would be LOADING for both, which we are changing to ABORTED.

Reply:
> It seems to me that as I mentioned there's at least a possibility that the image actually is loaded. This thread being interrupted doesn't stop the Media loading threads.

No, it doesn't.

>Since there's only one image for this id, LOADING and COMPLETE ought to be
mutually exclusive. In fact I'd expect only one of the flags to be set.

Right. At least I came to this conclusion: only one of the status constants is used although they're treated like flags in the code.

> So I can I suppose see the logic that this API returning ABORTED instead of LOADING
makes sense, but that is because this method removes it from the tracker ..

Since the image is removed from MediaTracker, it cannot update its status even if loading completes at a later time. This is why I think the API should return `ABORTED` instead of `LOADING`.


> …for Not-Interrupted case the result would be ERRORED for url link and COMPLETE for local Image,

Why? It could be `COMPLETE` for either. If the image from the URL is loaded successfully, its status will be `COMPLETE`.

> but for Interrupted case the result would be LOADING for both, which we are changing to ABORTED.

Again, not necessarily. In the test, the thread is interrupted before the image is added to MediaTracker. As soon as `waitForID` is called, it immediately throws `InterruptedException`. This is why the image status is likely to be `LOADING`.